addons/web/static/src/js/views/search_panel.js

```diff
@@ -389,19 +379,22 @@ const SearchPanel = Widget.extend({
     _fetchCategories(force) {
         const proms = [];
         for (const category of Object.values(this.categories)) {
+            const { enableCounters, expand, hierarchize } = category;
             const field = this.fields[category.fieldName];
-            if (force || category.enableCounters) {
+            if (force || enableCounters || !expand) {
```

Comment:
Can be change in fourth commit: there it is no more necessary to reload each time because we only use search_domain
and a test is done before to know when it changes. Good observation.

Reply:
Done in fourth commit